packages/drafts-realm/campaign-form.gts

```diff
@@ -0,0 +1,439 @@
+          @options={{this.campaignTypes}}
+          @dropdownClass='boxel-select-campaign-type'
+          as |item|
+        >
+          <div>{{item.name}}</div>
+        </BoxelSelect>
+      </FieldContainer>
+      <FieldContainer
+        @label='Parent Campaign'
+        data-test-field='parent_campaign'
+        @tag='label'
+        class='field'
+      >
+        <@fields.parent_campaign />
+      </FieldContainer>
+      <FieldContainer
+        @label='Description'
+        data-test-field='description'
+        @tag='label'
+        class='field'
+      >
+        <BoxelInput
+          @type='textarea'
+          @value={{this.args.model.description}}
+          @onInput={{this.updateDescription}}
+        />
+      </FieldContainer>
+      <FieldContainer
+        @label='Start Date'
+        data-test-field='start_date'
+        @tag='label'
+        class='field'
+      >
+        <BoxelInput
+          type='date'
+          @value={{this.selectedStartDate}}
+          @onInput={{fn this.parseDateInput 'start_date'}}
+          @max='9999-12-31'
+        />
+      </FieldContainer>
+      <FieldContainer
+        @label='End Date'
+        data-test-field='end_date'
+        @tag='label'
+        class='field'
+      >
+        <BoxelInput
+          type='date'
+          @value={{this.selectedEndDate}}
+          @onInput={{fn this.parseDateInput 'end_date'}}
+          @max='9999-12-31'
+        />
+      </FieldContainer>
+      <FieldContainer
+        @label='Num Sent in Campaign'
+        data-test-field='number_sent'
+        @tag='label'
+        class='field'
+      >
+        <@fields.number_sent />
+      </FieldContainer>
+      <FieldContainer
+        @label='Expected Response (%)'
+        data-test-field='expected_response_percentage'
+        @tag='label'
+        class='field'
+      >
+        <@fields.expected_response_percentage />
+      </FieldContainer>
+      <FieldContainer
+        @label='Expected Revenue in Campaign (RM)'
+        data-test-field='expected_revenue'
+        @tag='label'
+        class='field'
+      >
+        <@fields.expected_revenue />
+      </FieldContainer>
+      <FieldContainer
+        @label='Budgeted Cost in Campaign (RM)'
+        data-test-field='budgeted_cost'
+        @tag='label'
+        class='field'
+      >
+        <@fields.budgeted_cost />
+      </FieldContainer>
+      <FieldContainer
+        @label='Actual Cost in Campaign (RM)'
+        data-test-field='actual_cost'
+        @tag='label'
+        class='field'
+      >
+        <@fields.actual_cost />
+      </FieldContainer>
+    </div>
+    <style>
+      .campaign-form-edit {
+        display: grid;
+        gap: var(--boxel-sp-lg);
+        padding: var(--boxel-sp-xl);
+      }
+
+      .boxel-input-campaign-active {
+        grid-column: unset;
+      }
+    </style>
+  </template>
+}
+
+export class CampaignForm extends CardDef {
+  @field name = contains(StringField, {
```

Comment:
can you 
- add a display name
- change snake case to camel case

Reply:
Hi @tintinthong , please have a look again at the latest commit thanks!

- add a display name - it exists but now I place it after the class definition for better clarity
- change snake case to camel case - done!